### ISSUE TITLE
Circuit breaker pattern to prevent self-killing

### ISSUE THREAD

**Issue by sroze** [sroze/FaultTolerance#6]

**Comment by shouze** [comment]
ping @sroze is it still an issue to consider in this projet? If yes... do you know polly from .Net world? It brings some circuit breaker feature: https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/implement-circuit-breaker-pattern

**Comment by sroze** [comment]
@shouze this is not done yet but would be great to have such in Tolerance! Would you like to work on this a make a PR? :)

**Comment by dbalabka** [comment]
@sroze is this still actual? 

I did my implementation of Circuit Breaker from scratch with integration to Symfony, Guzzle and store metrics in APCu which is inspired by https://github.com/upwork/phystrix and https://github.com/Netflix/Hystrix (it is not publicly available because of company politics)

Probably, I can contribute if you are going to continue work on this project. I see that the latest release was on Oct 6, 2017

**Comment by llaville** [comment]
There is an other implementation alternative :  https://github.com/ackintosh/ganesha

**Comment by sroze** [comment]
That would be a great contribution indeed! We didn’t add much recently but
it’s definitely still “actual” :)

**Comment by sroze** [comment]
Indeed, I didn’t know Ganesha. It would probably be better to integrate
actually.
